client/my-sites/checkout/checkout-thank-you/redesign-v2/products/product-domain.tsx

```diff
@@ -0,0 +1,86 @@
+import { isDomainTransfer } from '@automattic/calypso-products';
+import formatCurrency from '@automattic/format-currency';
+import { Button, ClipboardButton } from '@wordpress/components';
+import { useI18n } from '@wordpress/react-i18n';
+import { translate } from 'i18n-calypso';
+import { useState } from 'react';
+import ThankYouProduct from 'calypso/components/thank-you-v2/product';
+import { domainManagementList, domainManagementRoot } from 'calypso/my-sites/domains/paths';
+import type { ReceiptPurchase } from 'calypso/state/receipts/types';
+
+type ProductDomainProps = {
+	purchase?: ReceiptPurchase;
+	domainName?: string;
+	shareSite?: boolean;
+	siteSlug?: string | null;
+	currency?: string;
+};
+
+const ProductDomain = ( {
+	purchase,
+	domainName,
+	shareSite,
+	siteSlug,
+	currency,
+}: ProductDomainProps ) => {
+	const { __ } = useI18n();
+	const [ isCopying, setIsCopying ] = useState( false );
+	const domain = domainName ?? purchase?.meta;
+
+	// Do not proceed if a domain is not specified by domain name or a purchase object.
+	if ( ! domain ) {
+		return null;
+	}
+
+	const handleShareSite = ( processing: boolean ) => () => {
+		setIsCopying( processing );
+	};
+
+	const purchaseLabel = ( priceInteger: number ) => {
+		if ( priceInteger === 0 ) {
+			return __( 'We’ve paid for an extra year' );
+		}
+
+		const priceFormatted = formatCurrency( priceInteger, currency ?? 'USD', {
+			stripZeros: true,
+			isSmallestUnit: true,
+		} );
+
+		return translate( '%(priceFormatted)s for one year', { args: { priceFormatted } } );
+	};
```

Comment:
This logic was taken from [`<DomainTransferredList>`](https://github.com/Automattic/wp-calypso/blob/2c06f4634f53f0e86d2a4528a7158f4e689c451f/client/my-sites/checkout/checkout-thank-you/redesign-v2/sections/product/DomainsTransferredList.tsx#L24-L38).

If the purchase is a domain transfer, we'll be rendering pricing info instead of action buttons. Note that, in this PR, we're only refactoring the domain-only congrats for non-transfers, so this logic won't kick in until we roll out the subsequent refactoring PR for transfers https://github.com/Automattic/wp-calypso/pull/86203